clippy_lints/src/methods/search_is_some.rs

```diff
@@ -146,3 +173,203 @@ pub(super) fn check<'tcx>(
+            let span = map.span(cmt.hir_id);
+            let start_span = Span::new(self.next_pos, span.lo(), span.ctxt());
+            let mut start_snip = snippet_with_applicability(self.cx, start_span, "..", &mut self.applicability);
+
+            if cmt.place.projections.is_empty() {
+                // handle item without any projection, that needs an explicit borrowing
+                // i.e.: suggest `&x` instead of `x`
+                self.suggestion_start.push_str(&format!("{}&{}", start_snip, ident_str));
+            } else {
+                // cases where a parent call is using the item
+                // i.e.: suggest `.contains(&x)` for `.find(|x| [1, 2, 3].contains(x)).is_none()`
+                if let Some(parent_expr) = get_parent_expr_for_hir(self.cx, cmt.hir_id) {
+                    if let ExprKind::Call(_, call_args) | ExprKind::MethodCall(_, _, call_args, _) = parent_expr.kind {
+                        let expr = self.cx.tcx.hir().expect_expr(cmt.hir_id);
+                        let arg_ty_kind = self.cx.typeck_results().expr_ty(expr).kind();
+
+                        if matches!(arg_ty_kind, ty::Ref(_, _, Mutability::Not)) {
+                            let start_span = Span::new(self.next_pos, span.lo(), span.ctxt());
+                            let start_snip =
+                                snippet_with_applicability(self.cx, start_span, "..", &mut self.applicability);
+
+                            // suggest ampersand if call function is taking args by ref
+                            let takes_arg_by_ref = self.func_takes_arg_by_ref(parent_expr, cmt.hir_id);
+
+                            // do not suggest ampersand if the ident is the method caller
+                            let ident_sugg =
+                                if !call_args.is_empty() && call_args[0].hir_id == cmt.hir_id && !takes_arg_by_ref {
+                                    format!("{}{}", start_snip, ident_str)
+                                } else {
+                                    format!("{}&{}", start_snip, ident_str)
+                                };
+                            self.suggestion_start.push_str(&ident_sugg);
+                            self.next_pos = span.hi();
+                            return;
+                        }
+
+                        self.applicability = Applicability::Unspecified;
+                    }
+                }
+
+                // handle item projections by removing one explicit deref
+                // i.e.: suggest `*x` instead of `**x`
+                let mut replacement_str = ident_str;
+
+                let mut projections_handled = false;
+                cmt.place.projections.iter().enumerate().for_each(|(i, proj)| {
+                    match proj.kind {
+                        // Field projection like `|v| v.foo`
+                        ProjectionKind::Field(idx, variant) => match cmt.place.ty_before_projection(i).kind() {
```

Comment:
I did a short test, and it seems like Rust doesn't care if `v` is a reference or not, as it only accesses the field. The implementation looks good to me, could you update the comment to say that field projections are handled by the compiler and don't need any adjustment from our side? :upside_down_face: